src/main/java/org/openrefine/extensions/commons/importer/FileRecordToRows.java

```diff
@@ -9,19 +9,42 @@
 
 public class FileRecordToRows implements TableDataReader {
     List<Object> rowsOfCells;
-
+    FileRecord fileRecord;
     Iterator<FileRecord> iteratorFileRecords;
+    int relatedCategoriesIndex = 0;
 
     public FileRecordToRows(Iterator<FileRecord> iteratorFileRecords) {
         this.iteratorFileRecords = iteratorFileRecords;
     }
+
     @Override
     public List<Object> getNextRowOfCells() throws IOException {
         rowsOfCells = new ArrayList<>();
-        rowsOfCells.add(iteratorFileRecords.next());
-
-        return rowsOfCells;
+        // check if there's rows remaining from a previous call
+        if (fileRecord != null && relatedCategoriesIndex < fileRecord.relatedCategories.size()) {
```

Comment:
The second condition might throw a `NullPointerException` if `fileRecord.relatedCategories == null`, it might be good to check for that.